php/Factory.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace PHPCD;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+use Lvht\MsgpackRpc\JsonMessenger;
+use Lvht\MsgpackRpc\Io;
+use Lvht\MsgpackRpc\MsgpackMessenger;
+
+/**
+ * Simple factory to separate details of object creation
+ */
+class Factory
+{
+    /**
+     * @return ContainerBuilder
+     */
+    public function createDIContainer($configFileName, $configDir, $additionalParameters = [])
+    {
+        $container = new ContainerBuilder;
+
+        $loader = new YamlFileLoader($container, new FileLocator($configDir));
+
+        $loader->load($configFileName);
+
+        foreach ($additionalParameters as $name => $value) {
+            $container->setParameter($name, $value);
+        }
+
+        $container->compile();
+
+        return $container;
+    }
+
+    /**
+     * @param string $implementation absolute namespace path to a concrete logger
+     * @param array $parameters parameters to logger's constructor
+     * @return \Psr\Log\LoggerInterface
+     */
+    public function createLogger($implementation, $parameters = [])
+    {
+        switch ($implementation) {
```

Comment:
I don't think we need so many Logger. As you prefer to implemnt our own `PHPCD\Log\DateTimeLogger`, the `Monolog\Logger` is become needless. 

All in all, I think we should remove `PHPCD\Log\Logger` and `Monolog\\Logger`